src/Modules/OrchardCore.Commerce/Views/OrderPart_Edit.cshtml

```diff
@@ -185,16 +185,29 @@
 <script at="Foot" depends-on="vuejs">
     const vueQuery = '#orderPart__products';
     let lineItems = @Json.Serialize(Model.LineItems);
-    let availableAttributes = JSON.parse('@Html.Raw(serializedDictionary)');
+    let availableAttributes = JSON.parse('@Html.Raw(serializedAttributesDictionary)');
 
     lineItems.forEach(item => {
         item.productUrl = '/Admin/Contents/ContentItems/' + item.productRouteValues.ContentItemId + '/Edit';
         item.availableAttributes = availableAttributes;
     })
 
-    var products = new Vue({
+    let products = new Vue({
         el: vueQuery,
-        data: { lineItems: lineItems },
+        data: {
+            availableCurrencyCodes: JSON.parse('@Html.Raw(serializedAvailableCurrencyCodes)'),
+            lineItems: lineItems,
+        },
+        computed: {
+            calculatedTotal() {
+                return this.lineItems
+                    .map(item => item.quantity * item.unitPriceValue)
+                    .reduce((total, value) => total + value, 0);
+            },
+            displayedCurrency() {
+                return this.lineItems.map(item => this.availableCurrencyCodes[item.unitPriceCurrencyIsoCode])[0];
```

Comment:
Why map the whole list? I think you want [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) (`?.` operator).

Reply:
> Why map the whole list?

To cook is to sometimes get lost in the sauce. Thanks, should be simpler now.